app/views/icons/show.haml

```diff
@@ -7,8 +7,8 @@
       &raquo;
       = link_to "#{@icon.user.username}'s Galleries", user_galleries_path(@icon.user)
   &raquo;
-  - if @icon.galleries.count == 1
-    = link_to @icon.galleries.first.name, gallery_path(@icon.galleries.first)
+  - if @galleries.present?
+    = link_to @galleries.first.name, url_for(@galleries.first)
```

Comment:
Won't this load-into-memory? I guess if it's using the association cache that doesn't matter but I vaguely recall a thing that queried exists if it wasn't loaded and the cache if it was.

Reply:
Yes, @galleries is loaded in memory, because galleries/single still uses Gallery objects. I could try to unravel it to one query for <all icon#show> and another to get Galleries for view=galleries if we wanted though?

Reply:
Yeah I'm mostly eying it because I have learned to be leery of `.present?` and we're only using the first one here afaict

Reply:
I mean I do literally at a later point fix a bad usage of `.present?` you are not wrong xD